src/components/Form/Form.jsx

```diff
@@ -0,0 +1,53 @@
+import { useState } from "react";
+import styles from "./Form.module.scss";
+
+const Form = ({
+  searchTerm,
+  setSearchTerm,
+  setPrevSearchTerm,
+  isFetchingData,
+  setIsFetchingData,
+  setSearchResults,
+}) => {
+  const dataFetch = async () => {
+    const searchQuery = searchTerm.replaceAll(" ", "+");
+    const response =
+      await fetch(`https://www.googleapis.com/books/v1/volumes?q=${searchQuery}&maxResults=40
+`);
+    const results = await response.json();
+    const resultsToArray = Object.values(results);
+    const arrayOfBooks = resultsToArray[2];
+    console.log("query results", arrayOfBooks);
+    setSearchResults(arrayOfBooks);
+    setIsFetchingData(false);
```

Comment:
that would allow something maybe a bit easier to handle that loading state like 
```js
const onSubmit = () => { 
  setIsFetchingData(true)
 searchForBooks(searchTerm)
    .then(results => setSearchResults(results))
    .catch(error => handleErrorState(error))
    .finally(() => setIsFetchingData(false))
} 
```